app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -114,8 +115,75 @@ def assigned
       render json: { topics: serialize_data(topics, AssignedTopicSerializer) }
     end
 
+    def group_assignments
+      if SiteSetting.assign_enabled?
+        offset = (params[:page].to_i * 30 || 0).to_i
+        is_group = (params[:is_group] || 1).to_i
+
+        if is_group == 1
+          topics = Topic
+            .includes(:tags)
+            .includes(:user)
+            .joins("JOIN topic_custom_fields tcf ON topics.id = tcf.topic_id AND tcf.name = 'assigned_to_id' AND tcf.value IS NOT NULL")
+            .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", params[:group_name])
+            .order('tcf.value::integer, topics.bumped_at desc')
+
+          load_more = topics.to_a.length > 30 * (params[:page].to_i + 1)
+
+          topics = topics
+            .offset(offset)
+            .limit(30)
+
+          users = User.where("users.id IN (SELECT group_users.user_id FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", params[:group_name])
+        else
+          users = User.where("username = ?", params[:group_name])
+
+          topics = Topic
+            .includes(:tags)
+            .includes(:user)
+            .joins("JOIN topic_custom_fields tcf ON topics.id = tcf.topic_id AND tcf.name = 'assigned_to_id' AND tcf.value IS NOT NULL")
+            .where("tcf.value::int = ?", users[0].id)
+            .order('tcf.value::integer, topics.bumped_at desc')
+
+          load_more = topics.to_a.length > 30 * (params[:page].to_i + 1)
+
+          topics = topics
+            .offset(offset)
+            .limit(30)
+
+        end
```

Comment:
There is lots of repetition in the `if` and `else` part of this statement. Can we move some stuff outside the if statement?

Reply:
Sure, it's now done as specified in the new commit😀